### PR TITLE
Advanced features refactor

### PR DESCRIPTION
**Pull request by Transurgeon**
## Description
Please include a short summary of the change.
This PR introduces some updates to the documentation, especially making many changes to the advanced features section of the tutorial. 
- moved SDP, MIP, attributes to a new section titled ``Advanced Constraints``
- moved solver related stuff to a new section titled ``Solver Features``
- moved DPP and sensitivity analysis to a new section titled ``DPP``
- combined license, related projects, citations and short course into a new top level page titled ``other``
- removed content tree in FAQ since the new theme does it automatically
- other minor changes that use the sphinx immaterial theme features (ex: using collapsible instead of dropdown)
The documentation changes can be previewed [here](https://cvxpy--49.org.readthedocs.build/en/49/).

## Type of change
- [ ] New feature (backwards compatible)
- [ ] New feature (breaking API changes)
- [ ] Bug fix
- [x] Other (Documentation, CI, ...)

## [Contribution checklist](https://www.cvxpy.org/contributing/index.html#contribution-checklist)
- [x] Add our license to new files.
- [x] Check that your code adheres to our coding style.
- [x] Write unittests.
- [x] Run the unittests and check that they’re passing.
- [x] Run the benchmarks to make sure your change doesn’t introduce a regression.

### CODE REVIEW

**Comment by SteveDiamond** [comment]
How is this going? Let me know if I can help! There is a merge conflict right now, not sure how to resolve it.

File: doc/source/tutorial/solvers/index.rst
Comment (Transurgeon) [inline doc/source/tutorial/solvers/index.rst:1]: This file includes all the solver options along with other sections from the advanced features related to solvers.

**Review by Transurgeon — COMMENTED** [review]
ping for review. @SteveDiamond @PTNobel

File: doc/source/tutorial/constraints/index.rst [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+        length of shape.
+    :type boolean: bool or list of tuple
+    :param integer: Is the variable integer? The semantics are the same as the boolean argument.
+    :type integer: bool or list of tuple
+    :param sparsity: Fixed sparsity pattern for the variable.
+    :type sparsity: list of tuplewith
+    :param pos: Is the variable constrained to be positive?
+    :type pos: bool
+    :param neg: Is the variable constrained to be negative?
+    :type neg: bool
```
Comment (SteveDiamond) [inline doc/source/tutorial/constraints/index.rst:57]: I think this is missing the new ``bounds`` field.

**Review by SteveDiamond — APPROVED** [review]
Looks great! I understand the refactoring now, thanks for fixing the merge conflict. I only noticed one small issue.

**Comment by Transurgeon** [comment]
@SteveDiamond I removed the RDT setup file. If there are no other issues with the PR, we are good for merging.

**Comment by rileyjmurray** [comment]
WOOOOOOOO!!!

**Pull request merged**